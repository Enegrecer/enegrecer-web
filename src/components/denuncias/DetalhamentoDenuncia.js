import PropTypes from 'prop-types'
import React, { Component } from 'react';
import { cortarPalavra } from '../../utils/helpers';
import { CampoTexto, RadioGrupoBotoes, Combobox } from '../FormGroups';
import { estados } from './estados'
import * as ConstantesCSS from '../layouts/ConstantesCss'
import * as Tela from '../../utils/materializeCSS'


export default class DetalhamentoDenuncia extends Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
    this.state = {
      detalhamento: '',
      dataOcorrencia: '',
      horaOcorrencia: '',
      idCategoria: '',
      endereco: '',
      estado: ''
    };
  }
  componentDidMount() {
    const comboEstado = Tela.getElementoPorId('estado');
    comboEstado.on('change', (e) => {
      this.handleChange(e.target.value, 'estado');
    });
  }

  handleChange(value, property) {
    this.setState(
      { [property]: value },
      () => this.props.handleChange(this.state)
    );
  }

  render() {
    return (
      <div>
        <h1>Nova Denúncia</h1>

        <div className="row">
          <CampoTexto
            id={'detalhamento'}
            label={'* Detalhamento'}
            maxLen={255}
            type={'text'}
            inputClasse={ConstantesCSS.CLASSES_TEXTAREA}
            divClasse={`${ConstantesCSS.CLASSES_DIV_INPUT} col s12`}
            onChange={(e) => { this.handleChange(cortarPalavra(e.target.value, 255), 'detalhamento') }}
          />
        </div>


        <div className="row">

          <RadioGrupoBotoes
            classes={'col s3'}
            id={'idCategoria'}
            onChange={e => this.handleChange(e.target.checked, 'idCategoria')}
            botoes={[{ id: 'radioInjuria', valor: 'injuria', label: 'Injúria' },
              { id: 'radioRacismo', valor: 'racismo', label: 'Racismo' }]}
          />

          <CampoTexto
            id={'dataOcorrencia'}
            label={'Data do ocorrido'}
            inputClasse={ConstantesCSS.CLASSES_INPUT}
            divClasse={`${ConstantesCSS.CLASSES_DIV_INPUT} col s4`}
            maxLen={0}
            type={'date'}
            onChange={(e) => { this.handleChange(e.target.value, 'dataOcorrencia') }}
          />

          <CampoTexto
            id={'horaOcorrencia'}
            label={'Hora do ocorrido'}
            maxLen={0}
            type={'time'}
            inputClasse={ConstantesCSS.CLASSES_INPUT}
            divClasse={`${ConstantesCSS.CLASSES_DIV_INPUT} col s5`}
            onChange={(e) => { this.handleChange(e.target.value, 'horaOcorrencia') }}
          />
        </div>

        <h4>Local do crime</h4>
        <div className="row">
          <CampoTexto
            id={'endereco'}
            label={'Endereço'}
            maxLen={255}
            type={'text'}
            inputClasse={ConstantesCSS.CLASSES_TEXTAREA}
            divClasse={`${ConstantesCSS.CLASSES_DIV_INPUT} col s12`}
            onChange={(e) => { this.handleChange(cortarPalavra(e.target.value, 255), 'endereco') }}
          />
        </div>
        <div className="row">
          <Combobox
            id={'estado'}
            value={this.state.estado}
            onChange={this.handleChange}
            itens={estados}
            divClasse={`${ConstantesCSS.CLASSES_DIV_INPUT} col s12`}
            label={'Selecione o Estado:'}
            valorPadrao={'Selecione'}
          />
        </div>
      </div>
    )
  }
}

DetalhamentoDenuncia.defaultProps = { handleChange: () => { } };

DetalhamentoDenuncia.propTypes = {
  handleChange: PropTypes.func
};
