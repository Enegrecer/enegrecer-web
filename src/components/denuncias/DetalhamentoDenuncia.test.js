import React from 'react';
import PropTypes from 'prop-types';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import { shallow, mount } from 'enzyme';
import DetalhamentoDenuncia from './DetalhamentoDenuncia';


describe('DetalhamentoDenuncia Component', () => {
  it('deve renderizar o componente sem erros', () => {
    const wrapper = shallow(<DetalhamentoDenuncia />);
    expect(wrapper.exists()).toBe(true);
  });
  describe('Renderizações dos campos do componente', () => {
    it('deve renderizar sem erro o campo de detalhamento do componente', () => {
      const wrapper = shallow(<DetalhamentoDenuncia />);
      expect(wrapper.find('#detalhamento').length).toEqual(1);
    });

    it('deve renderizar sem erro o campo de horaOcorrencia do componente', () => {
      const wrapper = shallow(<DetalhamentoDenuncia />);
      expect(wrapper.find('#horaOcorrencia').length).toEqual(1);
    });

    it('deve renderizar sem erro o campo de dataOcorrencia do componente', () => {
      const wrapper = shallow(<DetalhamentoDenuncia />);
      expect(wrapper.find('#dataOcorrencia').length).toEqual(1);
    });

    it('deve renderizar sem erro o campo de idCategoria do componente', () => {
      const wrapper = mount(<DetalhamentoDenuncia />, {
        context: {
          muiTheme: getMuiTheme(),
        },
        childContextTypes: {
          muiTheme: PropTypes.object.isRequired,
        },
      });
      expect(wrapper.find('input#radioInjuria').length).toEqual(1);
      expect(wrapper.find('input#radioRacismo').length).toEqual(1);
    });

    it('deve renderizar sem erro o campo de endereco do componente', () => {
      const wrapper = shallow(<DetalhamentoDenuncia />);
      expect(wrapper.find('#endereco').length).toEqual(1);
    });

    it('deve renderizar sem erro o campo seletor de estados do componente', () => {
      const wrapper = shallow(<DetalhamentoDenuncia />);
      expect(wrapper.find('#estado').length).toEqual(1);
    });
  })

  describe('quando o valor do campo for alterado', () => {
    const wrapper = shallow(<DetalhamentoDenuncia />);

    beforeEach(() => {
      const eventMock = {
        target: {
          value: 'valor detalhamento',
        },
      };

      wrapper.find('#detalhamento').simulate('change', eventMock);
    });

    it('altera o valor correspondente no state', () => {
      expect(wrapper.state().detalhamento).toEqual('valor detalhamento');
    });
  });

  it('inicializa o componente com o valor padrao para state', () => {
    const wrapper = shallow(<DetalhamentoDenuncia />);

    const expectedState = {
      detalhamento: '',
      dataOcorrencia: '',
      horaOcorrencia: '',
      idCategoria: '',
      estado: '',
      endereco: '',
    };
    expect(wrapper.state()).toEqual(expectedState);
  });
});

