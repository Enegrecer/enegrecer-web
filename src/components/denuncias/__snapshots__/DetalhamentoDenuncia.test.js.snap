// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetalhamentoDenuncia Component deve possuir estrutura definida  1`] = `
<div>
  <h1>
    Nova Denúncia
  </h1>
  <div
    className="row"
  >
    <nome
      divClasse="col s12 m6 l6"
      id="detalhamento"
      onChange={[Function]}
      state="detalhamento"
      validacoes={Array []}
    />
  </div>
  <div
    className="row"
  >
    <botoesRadio
      botoes={
        Array [
          Object {
            "label": "Injúria",
            "valor": "injuria",
          },
          Object {
            "label": "Racismo",
            "valor": "racismo",
          },
        ]
      }
      classes="col s3"
      id=""
      state="idCategoria"
    />
    <Data
      divClasse="input-field col s4"
      id="dataOcorrencia"
      label="Data do ocorrido"
      state="dataOcorrencia"
    />
    <campoTexto
      divClasse="input-field col s5"
      id="horaOcorrencia"
      inputClasse="validate"
      label="Hora do ocorrido"
      maxLen={0}
      type="time"
    />
  </div>
  <h4>
    Local do crime
  </h4>
  <div
    className="row"
  >
    <campoTexto
      divClasse="input-field col s12"
      id="endereco"
      inputClasse="materialize-textarea"
      label="Endereço"
      maxLen={255}
      state="endereco"
      type="text"
    />
  </div>
  <div
    className="row"
  >
    <comboEstado
      classes="col s12"
      divClasse=""
      id="estado"
      state="estado"
    />
  </div>
</div>
`;
